File: Frontend/src/app/models/event.ts
export interface Event {
  id: number;
  title: string;
  description: string;
  location: string;
  category: string;
  dates: Date;
  type: string;
  imageUrls: string[];
}

export interface EventCreate {
  title: string;
  description: string;
  location: string;
  category: string;
  dates: Date;
  type: string;
  imageUrl?: string;
}

File: Backend/JebIncubator.Api/Models/Entities/Event.cs
/*
** EPITECH PROJECT, 2025
** survivor
** File description:
** Event.cs
*/

using System.ComponentModel.DataAnnotations;

namespace JebIncubator.Api.Models.Entities
{
    public class Event
    {
        public int Id { get; set; }
        
        [Required]
        public string Title { get; set; } = string.Empty;
        
        public string Description { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public string Location { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty;
        public int? ExternalApiId { get; set; }
    }
}
